### PR TITLE
feat(github-runners): add macOS 12 large and 13 large

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Try to add macOS 12 large, and macOS 13 large runners to our code

## Context

It looks like we're missing the `large` variant for macOS 12 and 13. We should probably add those variants to our code, because the official readme mentions those. [^1]

Also read the discussion here:

- https://github.com/renovatebot/renovate/discussions/26960

[^1]: [`actions/runner-images` readme, available images](https://github.com/actions/runner-images?tab=readme-ov-file#available-images)

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/github-runners/index.ts
Comment (HonkingGoose) [inline lib/modules/datasource/github-runners/index.ts:15]: Maintainers, can you check if this is going to work? The tests are green, but that's because my code lacks new tests, and there any no macOS tests in the `lib/modules/datasource/github-runners/index.spec.ts` file...

> `@HonkingGoose:` Are we missing MacOS and Windows tests?
>
> > `@secustor`: You are right there are currently no test for that and there should probably ones. 
> > That said the logic here is very small, the biggest junk of is here: 
> >
> > https://github.com/renovatebot/renovate/blob/7681ad00fb83b4d3404bfce24a695a2bf605693d/lib/modules/manager/github-actions/extract.spec.ts#L390

Comment (viceice) [reply lib/modules/datasource/github-runners/index.ts:15]: it's safe, we use docker versioning
https://github.com/renovatebot/renovate/blob/79706e6e90e279872d2ca0299200a60674e21bd0/lib/modules/datasource/github-runners/index.ts#L40

**Review by viceice — APPROVED** [review]

File: lib/modules/datasource/github-runners/index.ts
Comment (viceice) [inline lib/modules/datasource/github-runners/index.ts:15]:
```suggestion
      { version: '13-large' },
```

it seems the labels are wrong?

https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners

File: lib/modules/datasource/github-runners/index.ts
Comment (viceice) [inline lib/modules/datasource/github-runners/index.ts:16]:
```suggestion
      { version: '13-xlarge' },
```

File: lib/modules/datasource/github-runners/index.ts
Comment (viceice) [inline lib/modules/datasource/github-runners/index.ts:19]:
```suggestion
      { version: '12-large' },
```
`macos-12-xlarge` doesn't exist

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/datasource/github-runners/index.ts
Comment (HonkingGoose) [reply lib/modules/datasource/github-runners/index.ts:15]: I was wondering about that too! 🙂

File: lib/modules/datasource/github-runners/index.ts
Comment (HonkingGoose) [inline lib/modules/datasource/github-runners/index.ts:15]: Do I need to wait until this PR is merged?

- https://github.com/renovatebot/renovate/pull/27241

**Review by viceice — APPROVED** [review]

**Review by viceice — COMMENTED** [review]
we should add `isDeprecated` to all known unsupported and deprecated versions.

https://github.com/renovatebot/renovate/blob/e6d088abb0cdf8a229d8f7e4b90b759670320f10/lib/modules/datasource/types.ts#L57

**Review by secustor — APPROVED** [review]

**Comment by HonkingGoose** [comment]
> we should add `isDeprecated` to all known unsupported and deprecated versions.
> 
> https://github.com/renovatebot/renovate/blob/e6d088abb0cdf8a229d8f7e4b90b759670320f10/lib/modules/datasource/types.ts#L57

See issue:

- #27290

**Pull request merged**
